homeassistant/components/dexcom/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""The Dexcom integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pydexcom import AccountError, Dexcom, SessionError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_UNIT_OF_MEASUREMENT, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_SERVER, DOMAIN, MG_DL, PLATFORMS, SERVER_OUS
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=180)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up configured Dexcom."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Dexcom from a config entry."""
+    try:
+        dexcom = await hass.async_add_executor_job(
+            Dexcom,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_SERVER] == SERVER_OUS,
+        )
+    except AccountError:
+        return False
+    except SessionError:
+        raise ConfigEntryNotReady
+
+    entry.add_update_listener(update_listener)
+    if not entry.options:
+        entry.options = {CONF_UNIT_OF_MEASUREMENT: MG_DL}
```

Comment:
We can't update the options directly. We have to use the config entries api, `hass.config_entries.async_update_entry`. The options attribute is a read only mapping proxy.